conda_build/inspect_pkg.py

```diff
@@ -155,61 +159,42 @@ def replace_path(binary, path, prefix):
         return "not found"
 
 
-def test_installable(channel="defaults"):
+def test_installable(channel: str = "defaults") -> bool:
     success = True
-    log = get_logger(__name__)
-    has_py = re.compile(r"py(\d)(\d)")
-    for platform in ["osx-64", "linux-32", "linux-64", "win-32", "win-64"]:
-        log.info("######## Testing platform %s ########", platform)
-        channels = [channel]
-        index = get_index(channel_urls=channels, prepend=False, platform=platform)
-        for _, rec in index.items():
-            # If we give channels at the command line, only look at
-            # packages from those channels (not defaults).
-            if channel != "defaults" and rec.get("schannel", "defaults") == "defaults":
-                continue
-            name = rec["name"]
+    for subdir in ["osx-64", "linux-32", "linux-64", "win-32", "win-64"]:
+        log.info("######## Testing subdir %s ########", subdir)
+        for prec in get_index(channel_urls=[channel], prepend=False, platform=subdir):
+            name = prec["name"]
             if name in {"conda", "conda-build"}:
                 # conda can only be installed in the root environment
                 continue
-            if name.endswith("@"):
+            elif name.endswith("@"):
                 # this is a 'virtual' feature record that conda adds to the index for the solver
                 # and should be ignored here
                 continue
-            # Don't fail just because the package is a different version of Python
-            # than the default.  We should probably check depends rather than the
-            # build string.
-            build = rec["build"]
-            match = has_py.search(build)
-            assert match if "py" in build else True, build
-            if match:
-                additional_packages = [f"python={match.group(1)}.{match.group(2)}"]
```

Comment:
Where did `additional_packages` go? Can you explain why is it being removed?

Reply:
As I understood (and from my testing) this doesn't make a difference. The package already defines its Python dependency so theres no need for us to also define it here.

This was actually prompted by the removal of the regex search on 183. Packages that do not follow the `pyNN` build string pattern fails (e.g. https://anaconda.org/conda-test/itsdangerous/files).

See `conda inpsect channels conda-test --test-installable`.